### PR TITLE
LDES outputs bug fix + add duals outputting + storage capres

### PR DESCRIPTION
**Pull request by gmantegna**

### CODE REVIEW

File: src/GenX.jl [hunk 1]
```diff
@@ -39,7 +39,7 @@ using Statistics
 
 # Uncomment if Gurobi or CPLEX active license and installations are there and the user intends to use either of them
 #using CPLEX
-#using Gurobi
+using Gurobi
```
Comment (cfe316) [inline src/GenX.jl:42]: Please remove this line - it'll cause trouble for anyone trying to build. Same with the add to Project.toml.

File: src/model/resources/storage/storage_all.jl [hunk 2]
```diff
@@ -9,6 +9,7 @@ function storage_all!(EP::Model, inputs::Dict, setup::Dict)
 
 	dfGen = inputs["dfGen"]
 	Reserves = setup["Reserves"]
+	CapacityReserveMargin = setup["CapacityReserveMargin"]
```
Comment (cfe316) [inline src/model/resources/storage/storage_all.jl:12]: Could you use lower case here for the variable name?

File: src/write_outputs/write_net_revenue.jl [hunk 3]
```diff
@@ -32,6 +32,16 @@ function write_net_revenue(path::AbstractString, inputs::Dict, setup::Dict, EP::
...
 	end
 
+	# Add capacity constraint dual
+	dfNetRevenue[!, "Capacity_constraint_dual"] = fill(NaN, nrow(dfNetRevenue))
+	for y in intersect(dfGen[dfGen.Max_Cap_MW.>0, :R_ID], 1:G)
+		dual_val = dual.(EP[:cMaxCap][y])
+		if setup["ParameterScale"] == 1
+			dual_val *= ModelScalingFactor # converting MM$/GW-yr to $/MW-yr
+		end
+		dfNetRevenue[y, "Capacity_constraint_dual"] = dual_val
```
Comment (cfe316) [inline src/write_outputs/write_net_revenue.jl:42]: Consider separating the `for` and `if`. Could you `*=` the column afterward?

Or look at the solutions in other files: above the for loop have `modelscalingfactor = setup["ParameterScale"] ? ModelScalingFactor : 1` and then always `* modelscalingfactor`.

Comment (cfe316) [reply src/write_outputs/write_net_revenue.jl:42]: Is there a reason why the intersection is necessary? `R_ID` is always in the set 1:G.

**Review by cfe316 — CHANGES_REQUESTED** [review]
I'm glad to see that you're addressing these issues.

Would it be possible to separate into different pull requests?
1 - fix bug
2- add or enhance new feature

File: src/write_outputs/write_net_revenue.jl (hunk 3 above)
Comment (gmantegna) [reply src/write_outputs/write_net_revenue.jl:42]: Yes it needs to be only the generators with a maximum capacity constraint.

**Pull request closed without merging**